web/pdf_document_properties.js

```diff
@@ -87,71 +113,71 @@ class PDFDocumentProperties {
    * @param {Object} pdfDocument - A reference to the PDF document.
    * @param {string} url - The URL of the document.
    */
-  setDocumentAndUrl(pdfDocument, url) {
+  setDocument(pdfDocument, url) {
+    if (this.pdfDocument) {
```

Comment:
> I think there is no real need to check for this here since this._reset() does the check too.

There's a very good reason, and it's that usually `setFileSize` is called before `setDocument`. If we'd just unconditionally call `reset` here, then that would complete remove the point of `setFileSize`.
Maybe some sort of comment might be in order here!?

> Moreover, shouldn't this.pdfDocument be private as well, i.e., this._pdfDocument, as it may only be set using this setter (like this._url)?

Hmm, I'm guessing that I probably didn't change that since `this.pdfDocument` is used in *a lot* of `web/` files. Actually, I'm wondering if I shouldn't just revert the naming in this patch to its prior state, since I'm not sure if marking these as "private" properties really adds anything?